### PR TITLE
Fix errors in libvirt PinFile

### PR DESCRIPTION
**Pull request by 14rcole**
Fix errors in Libvirt PinFile that prevent the PinFile from being used without template data, as well as syntax errors.

### CODE REVIEW

File: docs/source/examples/workspaces/libvirt/PinFile [hunk 1]
```diff
@@ -24,11 +24,8 @@ libvirt-new:
...
-            {% if storage is defined %}
-            additional_storage: {{ storage }}
-            {% else %}
-            additional_storage: 10G
-            {% endif %}
+            additional_storage: {{ storage | default('10G') }}
+            {% if user is defined %}
             cloud_config:
               users:
                 - name: {{ user.name | default('herlo') }}
```
Comment (p3ck) [inline docs/source/examples/workspaces/libvirt/PinFile]: can we change this to 'admin' now?

File: docs/source/examples/workspaces/libvirt/PinFile [hunk 2]
```diff
@@ -37,6 +34,16 @@ libvirt-new:
                   sudo: {{ user.sudo | default('ALL=(ALL) NOPASSWD:ALL') }}
                   ssh_import_id: {{ user.ssh_id | default('gh:herlo') }}
```
Comment (p3ck) [inline docs/source/examples/workspaces/libvirt/PinFile]: same

File: docs/source/examples/workspaces/libvirt/PinFile [hunk 3]
```diff
@@ -37,6 +34,16 @@ libvirt-new:
...
                   lock_passwd: {{ user.lock | default(true) }}
+            {% else %}
+            cloud_config:
+              users:
+                - name: herlo
+                  gecos: Clint Savage
+                  groups: wheel
```
Comment (p3ck) [inline docs/source/examples/workspaces/libvirt/PinFile]: again, 'admin', 'Admin User', etc...

File: docs/source/examples/workspaces/libvirt/PinFile [hunk 4]
```diff
@@ -135,10 +150,17 @@ libvirt-custom-img:
             cloud_config:
               virt_type: virt-customize
               root_password: {{ password | default("testrootpwd") }} # insecure plaintext password
+              {% if user is defined %}
               users:
                 - name: {{ user.name | default('herlo') }}
```
Comment (p3ck) [inline docs/source/examples/workspaces/libvirt/PinFile]: same..

File: docs/source/examples/workspaces/libvirt/PinFile (hunk 1 above)
Comment (14rcole) [reply docs/source/examples/workspaces/libvirt/PinFile]: change what, user.name?

**Comment by 14rcole** [comment]
[merge]

**Comment by abraverm** [comment]
@14rcole , thanks how did you catch that?

**Comment by 14rcole** [comment]
@abraverm I was testing something else with libvirt and couldn't run linchpin up

**Pull request merged**